src/backends/interflop-backend-vprec/common/vprec_tools.c

```diff
@@ -0,0 +1,325 @@
+/*****************************************************************************\
+ *                                                                           *\
+ *  This file is part of the Verificarlo project,                            *\
+ *  under the Apache License v2.0 with LLVM Exceptions.                      *\
+ *  SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception.                 *\
+ *  See https://llvm.org/LICENSE.txt for license information.                *\
+ *                                                                           *\
+ *                                                                           *\
+ *  Copyright (c) 2015                                                       *\
+ *     Universite de Versailles St-Quentin-en-Yvelines                       *\
+ *     CMLA, Ecole Normale Superieure de Cachan                              *\
+ *                                                                           *\
+ *  Copyright (c) 2018                                                       *\
+ *     Universite de Versailles St-Quentin-en-Yvelines                       *\
+ *                                                                           *\
+ *  Copyright (c) 2019-2021                                                  *\
+ *     Verificarlo Contributors                                              *\
+ *                                                                           *\
+ ****************************************************************************/
+#include <math.h>
```

Comment:
For VPREC, for t > 27 we cannot guarantee correct rounding. See Samuel A. Figueroa. “When is Double Rounding Innocuous?” In: SIGNUM
Newsl. 30.3 (1995), pp. 21–26. issn: 0163-5778. doi: 10.1145/221332.
221334. url: https://doi.org/10.1145/221332.221334.

Reply:
and page 158 in my habilitation manuscript. Could you please git rebase and edit the message log?